The reason pax-web-tomcat-bundle fails is:
```
java.lang.NoClassDefFoundError: javax/servlet/http/PushBuilder
	at java.lang.Class.getDeclaredConstructors0(Native Method)
	at java.lang.Class.privateGetDeclaredConstructors(Class.java:2671)
	at java.lang.Class.getConstructor0(Class.java:3075)
	at java.lang.Class.getConstructor(Class.java:1825)
	at org.apache.tomcat.util.modeler.modules.MbeansDescriptorsIntrospectionSource.isBeanCompatible(MbeansDescriptorsIntrospectionSource.java:172)
	at org.apache.tomcat.util.modeler.modules.MbeansDescriptorsIntrospectionSource.supportedType(MbeansDescriptorsIntrospectionSource.java:145)
	at org.apache.tomcat.util.modeler.modules.MbeansDescriptorsIntrospectionSource.initMethods(MbeansDescriptorsIntrospectionSource.java:265)
	at org.apache.tomcat.util.modeler.modules.MbeansDescriptorsIntrospectionSource.createManagedBean(MbeansDescriptorsIntrospectionSource.java:305)
	at org.apache.tomcat.util.modeler.modules.MbeansDescriptorsIntrospectionSource.execute(MbeansDescriptorsIntrospectionSource.java:82)
	at org.apache.tomcat.util.modeler.modules.MbeansDescriptorsIntrospectionSource.loadDescriptors(MbeansDescriptorsIntrospectionSource.java:73)
	at org.apache.tomcat.util.modeler.Registry.load(Registry.java:590)
	at org.apache.tomcat.util.modeler.Registry.findManagedBean(Registry.java:494)
	at org.apache.tomcat.util.modeler.Registry.registerComponent(Registry.java:619)
	at org.apache.catalina.util.LifecycleMBeanBase.register(LifecycleMBeanBase.java:126)
	at org.apache.catalina.util.LifecycleMBeanBase.initInternal(LifecycleMBeanBase.java:50)
	at org.apache.catalina.realm.RealmBase.initInternal(RealmBase.java:1060)
	at org.apache.catalina.util.LifecycleBase.init(LifecycleBase.java:122)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:155)
	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:742)
	at org.apache.catalina.core.StandardEngine.startInternal(StandardEngine.java:211)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:164)
	at org.apache.catalina.core.StandardService.startInternal(StandardService.java:415)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:164)
	at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:874)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:164)
	at org.ops4j.pax.web.service.tomcat.internal.TomcatServerWrapper.start(TomcatServerWrapper.java:688)
	at org.ops4j.pax.web.service.tomcat.internal.TomcatServerController.start(TomcatServerController.java:105)
	at org.ops4j.pax.web.service.internal.Activator.performConfiguration(Activator.java:555)
	at org.ops4j.pax.web.service.internal.Activator.updateController(Activator.java:445)
	at org.ops4j.pax.web.service.internal.Activator.lambda$updateServerControllerFactory$1(Activator.java:347)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)
Caused by: java.lang.ClassNotFoundException: javax.servlet.http.PushBuilder cannot be found by org.ops4j.pax.web.pax-web-tomcat-bundle_8.0.30
	at org.eclipse.osgi.internal.loader.BundleLoader.generateException(BundleLoader.java:562)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass0(BundleLoader.java:507)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:434)
	at org.eclipse.osgi.internal.loader.ModuleClassLoader.loadClass(ModuleClassLoader.java:174)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	... 37 more
```

--- a/jndi/jndi-url-itest/pom.xml
+++ b/jndi/jndi-url-itest/pom.xml
@@ -47,7 +47,7 @@
         <cm.version>3.2.0-v20070116</cm.version>
         <depends-maven-plugin.version>1.5.0</depends-maven-plugin.version>
         <javax.annotation-api.version>1.3.2</javax.annotation-api.version>
-        <javax.servlet-api.version>3.1.0</javax.servlet-api.version>
+        <jakarta.servlet-api.version>4.0.4</jakarta.servlet-api.version>
         <jndi-api.version>${jndi-api.dev-version}</jndi-api.version>
         <jndi-bundle.version>${jndi-bundle.dev-version}</jndi-bundle.version>
         <jndi-url-itest-biz.version>${jndi-url-itest-biz.dev-version}</jndi-url-itest-biz.version>
@@ -281,9 +281,9 @@
             <scope>test</scope>
         </dependency>
         <dependency>
-            <groupId>javax.servlet</groupId>
-            <artifactId>javax.servlet-api</artifactId>
-            <version>${javax.servlet-api.version}</version>
+            <groupId>jakarta.servlet</groupId>
+            <artifactId>jakarta.servlet-api</artifactId>
+            <version>${jakarta.servlet-api.version}</version>
             <scope>test</scope>
         </dependency>
         <dependency>

--- a/jndi/jndi-url-itest/src/test/java/org/apache/aries/jndi/itests/JndiUrlIntegrationTest.java
+++ b/jndi/jndi-url-itest/src/test/java/org/apache/aries/jndi/itests/JndiUrlIntegrationTest.java
@@ -151,7 +151,7 @@ public class JndiUrlIntegrationTest extends AbstractIntegrationTest {
                 // Bundles
                 mavenBundle("org.eclipse.equinox", "cm").versionAsInProject(),
                 mavenBundle("org.eclipse.osgi", "org.eclipse.osgi.services").versionAsInProject(),
-                mavenBundle("javax.servlet", "javax.servlet-api").versionAsInProject(),
+                mavenBundle("jakarta.servlet", "jakarta.servlet-api").versionAsInProject(),
                 mavenBundle("javax.annotation", "javax.annotation-api").versionAsInProject(),
 
                 mavenBundle("org.ops4j.pax.web", "pax-web-extender-war").versionAsInProject(),

